tck/src/main/java/org/eclipse/microprofile/lra/tck/TckLRATypeTests.java

```diff
@@ -241,53 +266,56 @@ private void resourceRequest(String path, boolean startLRA, int expectedStatus,
 
             assertEquals(testName.getMethodName() + ": Unexpected status", expectedStatus, response.getStatus());
 
             if (response.getStatus() == Response.Status.PRECONDITION_FAILED.getStatusCode()) {
                 // 412 errors should abort running the target method so skip the LRA check
                 lraCheckType = MethodLRACheck.NONE;
                 methodLRAShouldBeActive = false;
-                methodLRA = "";
+                methodLraId = "";
             } else {
-                methodLRA = response.readEntity(String.class);
+                methodLraId = response.readEntity(String.class);
             }
 
             switch (lraCheckType) {
                 case NOT_PRESENT:
-                    assertEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLRA,
-                            0, methodLRA.length());
+                    assertEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLraId,
+                            0, methodLraId.length());
                     break;
                 case EQUALS:
                     assertEquals(testName.getMethodName() + ": Resource method should have ran with the incoming LRA",
-                            incomingLRA, methodLRA);
+                            incomingLraId, methodLraId);
                     break;
                 case NOT_EQUALS:
                     assertNotEquals(testName.getMethodName() + ": Resource method should not have run with the incoming LRA",
-                            incomingLRA, methodLRA);
+                            incomingLraId, methodLraId);
                     break;
                 default:
                     break;
             }
 
             if (methodLRAShouldBeActive) {
                 // validate that the method ran with an LRA and that it is still active
-                assertNotEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLRA,
-                        0, methodLRA.length());
-                assertFalse(lraClient.isLRAFinished(URI.create(methodLRA)));
-                lraClient.closeLRA(methodLRA);
-            } else if (methodLRA.length() != 0) {
+                assertNotEquals(testName.getMethodName() + ": Resource method should not have run with an LRA: " + methodLraId,
+                        0, methodLraId.length());
+                assertFalse(lraClient.isLRAFinished(URI.create(methodLraId)));
+                lraClient.closeLRA(methodLraId);
+            } else if (methodLraId.length() != 0) {
                 // otherwise it should be finished
-                assertTrue(lraClient.isLRAFinished(URI.create(methodLRA)));
+                assertTrue(lraClient.isLRAFinished(URI.create(methodLraId)));
             }
 
             if (lra != null) {
                 lraClient.closeLRA(lra);
             }
+            
+            return methodLraId;
```

Comment:
I would place this at the end of the method because now it seems that method never returns something (also requires the variable definition to be outside of the try {} )
But personal opinion.

Reply:
If we go with that we should put methodLraId = null on line 318 I think and go with a single return statement?

Reply:
I can look at this again when I'll do final sweep/cleaning before Final release :)

Reply:
The method initialises and returns the return value in the try block.
The method returns null in the catch block.

ie the code is safe and is readable so what is the problem with it.

Reply:
No real problem, just strange that the method has no return statement as last statement (one have to look for the location where the actual return happens)

Reply:
Okay, but you would still need to look back to see where the actual assignments to the return value occur. Returning the value in situ is arguably more readable than setting values and returning at the end.